connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,246 @@
+                      items:
+                        type: string
+                    logs:
+                      type: array
+                      description: The LOG AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    request:
+                      type: object
+                      description: The HTTP request sent to the source API for this page
+                      required:
+                        - url
+                        - headers
+                      properties:
+                        url:
+                          type: string
+                          description: URL that the request was sent to
+                        body:
+                          type: object
+                          description: The body of the HTTP request, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP request
+                    response:
+                      type: object
+                      description: The response from the source API for this page
+                      required:
+                        - status
+                        - headers
+                      properties:
+                        status:
+                          type: integer
+                          description: The status of the response
+                        body:
+                          type: object
+                          description: The body of the HTTP response, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP response
+    StreamReadRequestBody:
+      type: object
+      required:
+        - definition
+        - stream
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+        stream:
+          type: string
+          description: Name of the stream to read
+        numSlices:
+          type: integer
+          description: Number of stream slices to read from the source
+          default: 1
+        numPages:
+          type: integer
+          description: Number of pages to read from the source
+          default: 1
+        state:
+          type: object
+          description: State blob to use for incremental streams
+    ConnectorDefinitionBody:
+      $ref: ../../../../airbyte-cdk/python/airbyte_cdk/sources/declarative/config_component_schema.json
+    StreamsListRequestBody:
+      type: object
+      required:
+        - definition
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+    StreamsListRead:
```

Comment:
is this how the server will share the schema (whenever we support schema detection)?

Reply:
I don't think so, because detecting the schema will require making an API call to the source.

This endpoint is meant to be very fast, as it will likely be called on every change to the yaml editor contents in order to keep the stream list up to date. I think for schema detection, we will likely just want to do that as part of the `read` API call, e.g. add another property to the return value of that function that contains the schema of the results